compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -190,6 +190,15 @@ pub fn check_crate(tcx: TyCtxt<'_>) {
         }
     });
 
+    // Make sure we actually have a value for static items, as they aren't cached in incremental.
+    // While we could just wait for codegen to invoke this, the definitions freeze below will cause
+    // that to ICE, because evaluating statics can create more items.
+    tcx.hir().par_body_owners(|item_def_id| {
+        if let DefKind::Static { .. } = tcx.def_kind(item_def_id) {
+            let _ = tcx.eval_static_initializer(item_def_id);
+        }
+    });
+
```

Comment:
Why is this block necessary? The previous calls `tcx.ensure().eval_static_initializer`.

Reply:
Just test code to replicate the bug I had initially, it should not land